vyper/codegen/expr.py

```diff
@@ -527,9 +532,16 @@ def parse_BinOp(self):
             return
 
         pos = getpos(self.expr)
-
-        left.typ = new_type_to_old_type(self.expr.left._metadata["type"])
-        right.typ = new_type_to_old_type(self.expr.right._metadata["type"])
+        left.typ = (
+            new_type_to_old_type(self.expr.left._metadata["type"])
+            if isinstance(self.expr.left, vy_ast.VyperNode)
```

Comment:
why would `self.expr.left` ever not be an instance of `VyperNode`?

Reply:
this is a snippet from `examples/auctions/blind_auction.vy` where it is already converted to `IRNode` 

```
@external
@payable
def bid(_blindedBid: bytes32):
    # Check if bidding period is still open
    assert block.timestamp < self.biddingEnd

    # Check that payer hasn't already placed maximum number of bids
    numBids: int128 = self.bidCounts[msg.sender]
    assert numBids < MAX_BIDS

    # Add bid to mapping of all bids
    self.bids[msg.sender][numBids] = Bid({
        blindedBid: _blindedBid,
        deposit: msg.value
        })
    self.bidCounts[msg.sender] += 1
```

Reply:
i don't think that should be possible. all AST nodes are supposed to be VyperNodes; if we are seeing some that are not VyperNodes, it's indicative of a potential problem elsewhere.